src/utils/number.js

```
@@ -0,0 +1,30 @@
+import CONSTANTS from 'appConstants'
+
+function collapseTrailingZeros(string) {
+  const isInteger = /\.0+$/
+  const trailingZerosPattern = /(?<=\.[0-9]*)0+$/
+
+  if (isInteger.test(string)) {
+    return string.replace(isInteger, '')
+  }
+
+  return trailingZerosPattern.test(string)
+    ? string.replace(trailingZerosPattern, '')
+    : string
+}
+
+
+export function roundToTheNthDecimal(number, maxDecimals = CONSTANTS.NUMBER_INPUT_DECIMALS, revertToNumber = true) {
+
+  const INVALID_VALUES = [undefined, null, '', false]
+
+  if (INVALID_VALUES.includes(number)) return number
+
+  let roundedNumber = typeof number !== 'number'
+    ? number
+    : number.toFixed(maxDecimals)
+
+  roundedNumber = collapseTrailingZeros(roundedNumber)
+
+  return revertToNumber ? parseFloat(roundedNumber) : roundedNumber
+}
```

A better version of `toFixed()` which has the issue of turning a numbers into strings and filling up missing deccimas with zeros.  This one fixes both.

---

any reason not to just have 3 args here? seems the object is excessive. when the name of the other argument is in the title of the function, I think it's fair the function expects a non-default second argument. 

the global constant to a default arg could be misleading, (I believe the global constant NUMBER_INPUT_DECIMALS belongs on the other side of things, where it is passed in, because its a constant that is there to control the view, while this function is generic and has nothing to do with the view. We definitely wouldn't want its default behavior changing in other places this function is being used just because someone changes this constant :O

---

With 2 default arguments an object makes it easier to overwrite defaults, e.g. when you want to overwrite just one and not touching the other, you have to pass undefined as an argument to the parameters if you have to skip them. Also I like to pass objects instead of plain variables when I use boolean flags. I think it is easier to read on the other side: 

``` round(myNumber, 3, { parseToNumber: false })```
is easier to read and understand imho than 
``` round(myNumber, 3, false)```

---

re: the global constant: You are right here. I'll change that.

---

i agree re: the options in an object - especially after reading some sandi this morning:
my vote would go to the one you mention there: `round(myNumber, 3, { parseToNumber: false })`, with the options passed as object and the first two directly

---

agreed.